mrmustard/physics/ansatze.py

```diff
@@ -401,6 +497,372 @@ def __and__(self, other: PolyExpAnsatz) -> PolyExpAnsatz:
         return self.__class__(As, bs, cs)
 
 
+class DiffOpPolyExpAnsatz(PolyExpBase):
+    r"""
+    The ansatz of the Fock-Bargmann representation.
+
+    Represents the ansatz function:
+
+        :math:`F(z) = \sum_i [\sum_k c^(i)_k \partial_y^k \textrm{exp}((z,y)^T A_i (z,y) / 2 + (z,y)^T b_i)|_{y=0}]`
+
+    with ``k`` being a multi-index. The matrices :math:`A_i` and vectors :math:`b_i` are
+    parameters of the exponential terms in the ansatz, and :math:`z` is a vector of variables.
+
+    .. code-block::
+
+        >>> from mrmustard.physics.ansatze import DiffOpPolyExpAnsatz
+
+        >>> A = np.array([[1.0, 0.0], [0.0, 1.0]])
+        >>> b = np.array([1.0, 1.0])
+        >>> c = np.array([1.0,2.0,3.0])
+
+        >>> F = PolyExpAnsatz(A, b, c)
```

Comment:
This is a different class

Reply:
If I try this
```
A = np.array([[1.0, 0.0], [0.0, 1.0]])
b = np.array([1.0, 1.0])
c = np.array([1.0,2.0,3.0])

ansatze = DiffOpPolyExpAnsatz(A,b,c)
```
I trigger the error that the batch sizes need to be the same.

Reply:
Thanks. This is updated in the new docs.